src/lib/features/metrics/client-metrics/metrics-service-v2.ts

```diff
@@ -113,25 +121,43 @@ export default class ClientMetricsServiceV2 {
         }
     }
 
-    async filterExistingToggleNames(toggleNames: string[]): Promise<string[]> {
+    async filterExistingToggleNames(toggleNames: string[]): Promise<{
+        validatedToggleNames: string[];
+        unknownToggleNames: string[];
+    }> {
+        let unknownToggleNames: string[] = [];
+
         if (this.flagResolver.isEnabled('filterExistingFlagNames')) {
             try {
                 const validNames = await this.cachedFeatureNames();
 
                 const existingNames = toggleNames.filter((name) =>
                     validNames.includes(name),
                 );
+                if (this.flagResolver.isEnabled('reportUnknownFlags')) {
+                    unknownToggleNames = toggleNames
+                        .filter((name) => !existingNames.includes(name))
+                        .slice(0, MAX_UNKNOWN_FLAGS);
+                }
                 if (existingNames.length !== toggleNames.length) {
                     this.logger.info(
                         `Filtered out ${toggleNames.length - existingNames.length} toggles with non-existing names`,
                     );
                 }
-                return this.filterValidToggleNames(existingNames);
+
+                const validatedToggleNames =
+                    await this.filterValidToggleNames(existingNames);
+
+                return { validatedToggleNames, unknownToggleNames };
```

Comment:
Do we need to return unknownToggleNames? I may find the answer later.

Reply:
In case you haven't found the answer, this is what ends up being registered. See: https://github.com/Unleash/unleash/pull/9837/files#diff-8989b03379ec73aa0cebd572e8d7e819ee39b2fccf0cf4a93769007c73fc5054R239